redaxo/src/core/lib/Fragment/Component/Alert.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Redaxo\Core\Fragment\Component;
+
+use Redaxo\Core\Fragment\Slot;
+use rex_fragment;
+
+class Alert extends rex_fragment
+{
+    private string $fileName = 'core/Component/Alert.php';
+
+    public function __construct(
+        /**
+         * The alert's main content.
+         */
+        public Slot $slotDefault,
+
+        /**
+         * An icon to show in the alert.
+         */
+        public ?Slot $slotIcon = null,
+
+        /**
+         * Indicates whether or not the alert is open. You
+         * can toggle this attribute to show and hide the
+         * alert, or you can use the show() and hide()
+         * methods and this attribute will reflect the
+         * alert's open state.
+         */
+        public bool $open = false,
+
+        /**
+         * Enables a close button that allows the user to
+         * dismiss the alert.
+         */
+        public bool $closeable = false,
+
+        /**
+         * The alert's type.
+         */
+        public ?AlertType $type = null,
+
+        /**
+         * The length of time, in milliseconds, the alert
+         * will show before closing itself. If the user
+         * interacts with the alert before it closes
+         * (e.g. moves the mouse over it), the timer will
+         * restart. Defaults to Infinity, meaning the alert
+         * will not close on its own.
+         */
+        public ?int $duration = null,
+
+        /** @var array<string, string>|null */
+        public ?array $attributes = null,
+    ) {
+        parent::__construct([]);
+    }
+
+    public function parse($filename = null): string
```

Comment:
gibts einen use-case um eine bestehende komoponente mit einem externen fragment namen zu verwenden?
(die fragmente sind ja ohnehin bereits ohne diesen parameter überschreibbar via datepfade, wie wir es von fragmenten kennen)

Reply:
Ich wollte hier die zugehörige Datei an die Komponente fest machen, ohne dass bei jedem Verwenden der Dateipfad angeben werden muss.

Falls man das Fragment überschreiben mag, muss man in seinem AddOn nur denselben Dateifad anlegen.

Beispiel
`src/core/fragments/core/Component/Button.php`

Überscheibbar via
`src/addons/project/fragments/core/Component/Button.php`

Reply:
> Ich wollte hier die zugehörige Datei an die Komponente fest machen

seh ich auch so. somit ist der `$filename` parameter obsolete?

Reply:
> somit ist der $filename parameter obsolete?

Ja. Ich habe die Methode in `render` umbenannt. Weshalb?

1. der Param hätte wegen Kompatibilität zu `rex_fragment` erst einmal bleiben müssen
2. Passt finde ich besser :)